psi4/src/psi4/libmints/matrix.cc

```diff
@@ -673,17 +673,20 @@ void Matrix::set_block(const Slice &rows, const Slice &cols, const Matrix& block
     if (cols.end() - cols.begin() != block.colspi()) {
         throw PSIEXCEPTION("Invalid call to Matrix::set_block() column Slice doesn't match block's columns dimension.");
     }
-    const Dimension &rows_begin = rows.begin();
-    const Dimension &cols_begin = cols.begin();
-    Dimension block_rows = rows.end() - rows.begin();
-    Dimension block_cols = cols.end() - cols.begin();
+    if (symmetry_ != block.symmetry()) {
+        throw PSIEXCEPTION("Invalid call to Matrix::set_block() Target and destination matrix have different symmetry..");
+    }
+    const auto &rows_begin = rows.begin();
+    const auto &cols_begin = cols.begin();
+    auto block_rows = rows.end() - rows.begin();
+    auto block_cols = cols.end() - cols.begin();
     for (int h = 0; h < nirrep_; h++) {
         int max_p = block_rows[h];
-        int max_q = block_cols[h];
+        int max_q = block_cols[h ^ symmetry_];
         for (int p = 0; p < max_p; p++) {
             for (int q = 0; q < max_q; q++) {
                 double value = block.get(h, p, q);
-                set(h, p + rows_begin[h], q + cols_begin[h], value);
+                set(h, p + rows_begin[h], q + cols_begin[h ^ symmetry_], value);
```

Comment:
So previous behavior was the matrix would be set to wrong values if it wasn't symmetric?

Reply:
That or segfaulting.

Reply:
Our EOM-CC codes have LOTS of the "h^irrep" motif in them.  Before the story is lost to history, we should press TDC and CDS if they remember the source of the innovation.  I think it went back beyond Curt and Ed, so maybe back to psi1!?

Reply:
That traces back to the MOLECULE code. See Section 4 of [this report that TDC has saved](https://drive.google.com/file/d/1VtVpwvaleR1FF1hm2LchS2SQZ8_RHsJA/view). Molecule - A Program System for Non-Empirical Calculation of the Electronic Structure of Molecules II. Integral Section. University of Stockholm Institute of Physics Report 74 - 29, December 1974.

Hat tip to TDC for remembering this.